docs/stream.md

```diff
@@ -114,6 +116,39 @@ Argument     | Type                 | Required | Description
 
 ---
 
+##### Stream.scan
+
+Creates a new stream with the results of calling the function on every value in the stream with an accumulator and the incoming value.
+
+`stream = Stream.scan(fn, accumulator, stream)`
+
+Argument      | Type                             | Required | Description
+------------- | -------------------------------- | -------- | ---
+`fn`          | `(accumulator, value) -> result` | Yes      | A function that takes an accumulator and value parameter and returns a new accumulator value
+`accumulator` | `any`                            | Yes      | The starting value for the accumulator
+`stream`      | `Stream`                         | Yes      | Stream containing the values
+**returns**   | `Stream`                         |          | Returns a new stream containing the result
+
+[How to read signatures](signatures.md)
+
+---
```

Comment:
I'm not sure what mithril's signatures support, but it would be cool to say that accumulator isn't any, but instead its `<T>` or `a`

So in Hindley Milner

```
scan :: ( (a,b) -> a, a, Stream b ) -> Stream a`
```

Or in Typescript parlance:

```
<T,U> ( (a:T, b: U) -> T , initial: T, Stream<U>)
```

( Sorry @spacejack that TS signature probably isn't spot on )

I think its important to say, the accumulator's type will also be the type of the resulting stream, and the type of the first argument in the reducer function.